R/graded.R

```diff
@@ -90,3 +220,29 @@ legacy_graded <- function(...) {
     graded(...)
   )
 }
+
+get_from_env <- function(x, env, .caller) {
+  tryCatch(
+    get(x, envir = env),
+    error = function(e) {
+      label <- get0(".label", env, ifnotfound = NULL)
+      label <- if (!is.null(label)) paste0("In exercise `", label, "`: ")
+      not_found <- sprintf(
+        gettext("object %s not found", domain = "R"), 
+        paste0("'", x, "'")
+      )
+      if (grepl(not_found, e$message, fixed = TRUE)) {
```

Comment:
Is there another type of error that can be thrown?

Reply:
Hmmm, maybe? Inside `*-error-check`, `.result` might be the condition object from the (expected) user generated error. Will `tryCatch()` re-catch that error? I don't think so...

```r
env <- new.env()
tryCatch(
  stop("boom"),
  error = function(e) {
    env$.result <- e
  }
)

tryCatch(
  get(".result", envir = env),
  error = function(e) {
    message("Re-caught the condition object!")
  }
)
#> <simpleError in doTryCatch(return(expr), name, parentenv, handler): boom>
```